overlord/snapstate/snapmgr.go

```diff
@@ -385,10 +386,50 @@ func (m *SnapManager) blockedTask(cand *state.Task, running []*state.Task) bool
 	return false
 }
 
+// ensureUbuntuCoreTransition will migrate systems that use "ubuntu-core"
+// to the new "core" snap
+func (m *SnapManager) ensureUbuntuCoreTransition() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	var snapst SnapState
+	err := Get(m.state, "ubuntu-core", &snapst)
+	if err == state.ErrNoState {
+		return nil
+	}
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+
+	// check that there is no change in flight already
+	for _, chg := range m.state.Changes() {
```

Comment:
I think we want to make sure there are no changes around already linking/unlinking ubuntu-core (and core)